modules/database/funcs.go

```diff
@@ -229,3 +231,30 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+func GetServiceDbByName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*mariadbv1.MariaDBDatabase, error) {
```

Comment:
This makes our Database abstraction leaky. Can we move the finalizer handling to the Database object completely?

Reply:
As the client will use DeleteFinalizer, you don't need to export his function any more so please make it private by renaming it to getServiceDbByName

Reply:
@pinikomarov I thought you wanted to use GetServiceDbByName to return an initialized Database, so that you could do this in the cleanup:
db := GetServiceDbByName()
db.DeleteFinalizer()

instead of calling 
db := NewDatabase()
db.DeleteFinalizer()

Reply:
Ahh OK that can also be an option. So GetServiceDbByName() would return a *Database instead of *MariaDBDatabase. That works for me. But please rename GetServiceDbByName() to  GetDatabaseByName(). i) a service might have multiple DBs so the ServiceDb sounds strange. ii) this would made the name symmetric with NewDatabase.